Fix sensor configuration for disco_l475_iot1

erwango opened the pull request:
During dev for enabling sensors on board disco_l475_iot1 some changes gave been made on 
HTS221 and LIS3MDL configuration.
It introduced and unwanted dependency to the board when using the sensor.

This series fixes this issue.

erwango committed:
boards: disco_l475_iot1: fix method to configure sensor triggers

Defining TRIGGER_MODE choices in board Kconfig.defconfig files
brings unwanted dependency and warning messages when using sensors
on other boards.
Change sensor configuration by enabling wanted trigger choice rather
than redefining choice.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
Revert "sensors: lis3mdl: provide name to Kconfig choice"

This reverts commit 2e20f49167c09c7ffe6567395d40ac72bd88bd0b.

"choice" section in board Kconfig.default introduces a dependency
from the sensor to the board. Then a warning was generated
whenever the sensor is used on another board

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
Revert "sensors: hts221: provide name to Kconfig choice"

This reverts commit 2a7efb74191149b9d01e2cc1416c35c646253f4f.

"choice" section in board Kconfig.defconfig introduces a dependency
from the sensor to the board. As a consequence a warning was generated
whenever the sensor was used on another board.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

galak reviewed (changes requested):
Can you add in the 'revert' commits a short explanation as to why the revert.

erwango committed:
sample: fix configuration for sensor HTS221 sample application

Since no GPIO trigger is configured in hts221 sample application,
update prj.conf to enable TIGGER_NONE flag


Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>
